libsolidity/interface/OptimiserSettings.h

```diff
@@ -47,7 +47,7 @@ struct OptimiserSettings
 			"gvif"                     // Run full inliner
 			"CTUcarrLsTOtfDncarrIulc"  // SSA plus simplify
 		"]"
-		"jmuljuljul VcTOcul jmul";     // Make source short and pretty
+		"jmuljuljul VcTOcul jmulN";     // Make source short and pretty
```

Comment:
Why do we want to run `N` it at start and end? Isn't one of them enough?

Reply:
Some steps might remove functions and thus free up some possibilities for replacements that were not unique before.

Reply:
I assumed this would be similar `VarNameCleaner` and only needs to be run once at the end. I did test by removing the first `N`, and it had no difference in our tests; maybe the first `N` is redundant.

Reply:
Since it removes AST IDs, if you run it first, it can make the whole process more deterministic.